apps/internal/oauth/ops/accesstokens/tokens.go

```diff
@@ -173,14 +174,80 @@ type TokenResponse struct {
 	FamilyID       string                    `json:"foci"`
 	IDToken        IDToken                   `json:"id_token"`
 	ClientInfo     ClientInfo                `json:"client_info"`
-	ExpiresOn      internalTime.DurationTime `json:"expires_in"`
+	ExpiresOn      internalTime.DurationTime `json:"-"`
 	ExtExpiresOn   internalTime.DurationTime `json:"ext_expires_in"`
 	GrantedScopes  Scopes                    `json:"scope"`
 	DeclinedScopes []string                  // This is derived
 
 	AdditionalFields map[string]interface{}
+	scopesComputed   bool
+}
+
+func (tr *TokenResponse) UnmarshalJSON(data []byte) error {
+	type Alias TokenResponse
+	aux := &struct {
+		ExpiresIn json.Number `json:"expires_in,omitempty"`
+		ExpiresOn interface{} `json:"expires_on,omitempty"`
+		*Alias
+	}{
+		Alias: (*Alias)(tr),
+	}
+
+	// Unmarshal the JSON data into the aux struct
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	parseDuration := func(num json.Number) (int64, error) {
+		if num == "" {
+			return 0, nil
+		}
+		return num.Int64()
+	}
+
+	// Function to parse different date formats
+	parseExpiresOn := func(expiresOn string) (time.Time, error) {
```

Comment:
Pls add a comment pointing at https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4963 to explain when this can occur.

Reply:
Maybe I'm overlooking it, but that issue doesn't mention the platform returning the unsupported format

Reply:
The CX who reported this didn't make it very clear, it seems to be container + App Service.

Reply:
It would be good to know what the linked PR added support for so that support is easier to maintain. Setting that aside, the linked PR added ISO 8601 parsing. Why does this PR have two additional formats?

Reply:
We asked, the CX didn't answer. Ultimately, Azure SDK for .NET did not have this problem so it was clearly a regression. 

Not objecting to supporting only ISO 8601. The .NET code uses a general-purpose DateTime parser btw https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4964/files#diff-a7d2338916806e48fb637fd9caad6095fec8537a96d157bc2ea47ab64903fef9R93

I assume Azure.SDK for .NET did the same.